code/workspaces/auth-service/src/userManagement/authZeroUserManagement.js

```diff
@@ -18,30 +20,57 @@ const authKeyMapping = {
   user_id: 'userId',
 };
 
-export function asyncGetUsers() {
-  return requestAccessToken(accessTokenRequest())
-    .then(bearer => axios.get(`${authZeroManagementApi}/users`, {
-      params: {
-        fields: 'name,user_id',
-        per_page: '100',
-      },
-      headers: {
-        Authorization: `Bearer ${bearer}`,
-      },
-    }).then(extractUsers)
-      .then(processUsers)
-      .catch(() => {
-        throw new Error('Unable to retrieve users from User Management Service.');
-      }))
-    .catch((err) => {
+export async function asyncGetUsers() {
+  try {
+    try {
+      let page = 0;
+      let users = [];
+      let extractedUsers = [];
+      let activeUsers = [];
+      let allUsers = [];
+
+      const tokenRequest = await accessTokenRequest();
+      const bearer = await requestAccessToken(tokenRequest);
+
+      // Below approach is required as the auth0 API results are paginated, with a maximum
+      // overall limit, as well as a maximum per_page limit. This approach (non-async) is
```

Comment:
I don't quite understand the `non-async` part of this as it is still calling things in an asyn manner and is also an asyc function. Is it the fact that it doesn't request all of the pages at the same time?

Reply:
Ahhh that makes sense, thanks for clarifying!